wp-admin/js/widgets/media-widgets.js

```diff
@@ -258,95 +368,36 @@ wp.mediaWidgets = ( function( $ ) {
-			} );
+			wp.media.view.l10n.insertIntoPost = control.l10n.add_to_widget;
 
-			/**
-			 * Library which persists the customized display settings across selections.
-			 *
-			 * @class
-			 */
-			CustomizedDisplaySettingsLibrary = wp.media.controller.Library.extend( {
-
-				/**
-				 * Sync changes to the current display settings back into the current customized
-				 *
-				 * @param {Backbone.Model} displaySettings Modified display settings.
-				 * @returns {void}
-				 */
-				handleDisplaySettingChange: function handleDisplaySettingChange( displaySettings ) {
-					customizedDisplaySettings.set( displaySettings.attributes );
-				},
-
-				/**
-				 * Get the display settings model.
-				 *
-				 * Model returned is updated with the current customized display settings,
-				 * and an event listener is added so that changes made to the settings
-				 * will sync back into the model storing the session's customized display
-				 * settings.
-				 *
-				 * @param {Backbone.Model} model Display settings model.
-				 * @returns {Backbone.Model} Display settings model.
-				 */
-				display: function getDisplaySettingsModel( model ) {
-					var display;
-					display = wp.media.controller.Library.prototype.display.call( this, model );
-
-					display.off( 'change', this.handleDisplaySettingChange ); // Prevent duplicated event handlers.
-					display.set( customizedDisplaySettings.attributes );
-					if ( 'custom' === customizedDisplaySettings.get( 'link_type' ) ) {
-						display.linkUrl = customizedDisplaySettings.get( 'link_url' );
-					}
-					display.on( 'change', this.handleDisplaySettingChange );
-					return display;
-				}
-			} );
+			// Use our Post frame.
+			wp.media.view.MediaFrame.Post = control.CustomMediaFramePost;
 
 			mediaFrame = wp.media( {
-				frame: 'select',
-				button: {
-					text: control.l10n.add_to_widget
-				},
-				states: new CustomizedDisplaySettingsLibrary( {
-					library: wp.media.query( {
-						type: control.mime_type
-					} ),
-					title: control.l10n.select_media,
-					selection: selection,
-					multiple: false,
-					priority: 20,
-					display: true, // Attachment display setting.
-					filterable: false
-				} )
+				frame: 'post',
+				text: control.l10n.add_to_widget
 			} );
 
 			// Handle selection of a media item.
-			mediaFrame.on( 'select', function() {
-				var attachment;
+			mediaFrame.on( 'close', function() {
+				var attachment, state = mediaFrame.state();
+
+				// Restore the original wp.media.view.MediaFrame.Post object and language.
+				wp.media.view.MediaFrame.Post = control.originalMediaFramePost;
+				wp.media.view.l10n.insertIntoPost = control.originalButtonLanguage;
 
 				// Update cached attachment object to avoid having to re-fetch. This also triggers re-rendering of preview.
-				attachment = mediaFrame.state().get( 'selection' ).first().toJSON();
+				attachment = 'embed' === state.get( 'id' ) ? state.props.toJSON() : state.get( 'selection' ).first().toJSON();
```

Comment:
It happens if I open the media modal and then close it without having selected anything.